### PR TITLE
OpenClinica broken into bite-size commits

### PR DESCRIPTION
**Pull request by kaapstorm**
[OpenClinica](https://openclinica.com/) is a platform for managing clinical trial data. This PR allows CommCare forms to be transformed into an [ODM-formatted](http://www.cdisc.org/odm) (XML) document, which OpenClinica can import, using a custom report and export. 

In a CommCare app for clinical studies, the case type would be a study subject. In OpenClinica a study is divided into events which occur to subjects. Examples might be a visit, or taking medication. Data for an event is split up across one or more forms. Forms are made up of "items" (questions), which are grouped into "item groups" (question groups). We build up an ODM document by mapping CommCare questions to OpenClinica items, creating events based on the start and end times of CommCare forms, and associating them with OpenClinica subjects that correspond to CommCare subject cases.

Some data needs to be created in OpenClinica and cannot be imported in an ODM document: users; subjects; and events. This last one is particularly painful, because it means that a user will have to look at the report in CommCare to see a list of events for each subject, and then schedule those events manually in OpenClinica, before being able to import the data. Future integration will hopefully use OpenClinica web services via MOTECH, which will avoid this extra work because events can be scheduled using web services.

I've tried to break this PR into bite-size chunks for review. The most important commit is probably [7cb8c27](https://github.com/dimagi/commcare-hq/commit/7cb8c27e0a376cc5e298346666c5d51c07a89dbd) but you may want to go through the previous commits to understand how we get there.

This what the ODM Export report looks like, showing the Events to be scheduled in OpenClinica:
oc_odm_export_report

(That export button says "Export to Excel". I will be fixing that shortly to say "Export to OpenClinica".)

And this is the same data exported as an ODM document and imported into OpenClinica:
oc_successful_import

(Spot the typo? Yep, I wrote "Cantopril" instead of "Captopril". Oops.)

cc @NoahCarnahan and anyone else who has the time and enthusiasm to dive into this one.

### CODE REVIEW

File: custom/openclinica/const.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# CommCare case type of OpenClinica study subjects
+CC_SUBJECT_CASE_TYPE = 'subject'
+
+# Names of required case properties
+CC_SUBJECT_KEY = 'name'
+CC_STUDY_SUBJECT_ID = 'screening_number'
+
+# Include audit logs in export document
+AUDIT_LOGS = True
+
```
Comment (dimagimon) [inline custom/openclinica/const.py]: W391 blank line at end of file

File: custom/openclinica/management/commands/map_questions_to_openclinica.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+from collections import defaultdict
+from lxml import etree
+import os
+from django.conf import settings
+from django.core.management import BaseCommand
+import sys
```
Comment (dimagimon) [inline custom/openclinica/management/commands/map_questions_to_openclinica.py]: F401 'sys' imported but unused

**Comment by czue** [comment]
@kaapstorm guessing you should ping someone if you want review ;)

**Comment by kaapstorm** [comment]
@czue, I think you're right, but I feel a bit bad picking people because this is quite big.

So I'm pinging a few, @orangejenny, @proteusvacuum, @nickpell, and if any of you don't have time for this just say so.

**Comment by snopoke** [comment]
Seems generally OK though I do have some concerns about performance.

**Comment by kaapstorm** [comment]
:white_check_mark:

**Pull request merged**